dotnet-algorand-sdk/Transaction.cs

```diff
@@ -314,7 +311,7 @@ public byte[] lease
         /// <param name="freeze">account which can freeze or unfreeze holder accounts</param>
         /// <param name="clawback">account which can issue clawbacks against holder accounts</param>
         private Transaction(Address sender, ulong? fee, ulong? firstValid, ulong? lastValid, byte[] note,
-                           string genesisID, Digest genesisHash, BigInteger? assetTotal, int assetDecimals, bool defaultFrozen,
+                           string genesisID, Digest genesisHash, ulong? assetTotal, int assetDecimals, bool defaultFrozen,
```

Comment:
so it looks like it was ulong originally and we'd changed it to Bigint in an earlier stage. Why the switch back?

Reply:
the original (manual-improvements) implementation failed to create the transactions. It would try to write the BigIntiger as bytes and then failed to read it back. As there's no proper serializer defined for BigIntiger.